### PR TITLE
Add functionality to calculator

### PR DESCRIPTION
**Pull request by erickma1**
In this milestone, I did the following;

- Implement the event handlers to use the math logic from calculate.js and operate.js in the Calculator component.
- Use React life cycle methods

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,44 +1,59 @@
...
+  }
+
+  handleClick(item) {
+    const funcBtn = item.target.textContent;
+    this.setState((e) => calculate(e, funcBtn));
   }
 
   render() {
+    const { total, next, operation } = this.state;
+
```
Comment (rloterh) [inline src/components/Calculator.js]: - [ ] Let's improve the construction of the Calculator component by handling states with `useState` hook, which is more efficient in managing state of functional component. Our code should look like the following example.

```jsx
// ./src/components/Calculator.js

import React, { useState } from 'react';
import calculate from ../logic;

function Calculator() {
    const [total, setTotal] = useState(0);
    return {
      <div>
       // Calculator app 
      </div>
    };
}
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Eric, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🚦 _ 🏡 🏃‍♂️ 🏚️ 🌴 🗼

## Highlights 👍  
- Properly implemented Calculator logic!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
-  Let's make the ReadMe stand out even better by including instructions under the _Getting Started_ section, on how to run the app.
- Let's improve the look of the ReadMe by removing the circled:
- 
image

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by erickma1** [comment]
- Improve the construction of the Calculator component by handling states with useState hook
- Update readme template

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Eric,  

**Math magicians: events** task is complete! 🏁 Great job! Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on implementing recommended changes!
🎲 Event handling properly implemented!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
Let's make the ReadMe stand out even better by including instructions under the Getting Started section, on how to run the app.

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**